CIScripts/Tests/VRouterAgentTests.ps1

```diff
@@ -234,9 +355,40 @@ function Test-VRouterAgentIntegration {
         Assert-IsOnlyOnePkt0Injected -Session $Session
     }
 
-    Test-InitialPkt0Injection -Session $Session -TestConfiguration $TestConfiguration
-    Test-Pkt0RemainsInjectedAfterAgentStops -Session $Session -TestConfiguration $TestConfiguration
-    Test-OnePkt0ExistsAfterAgentIsRestarted -Session $Session -TestConfiguration $TestConfiguration
+    function Test-Pkt0ReceivesTrafficAfterAgentIsStarted {
+        Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
+               [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
+
+        Write-Host "===> Running: Test-Pkt0ReceivesTrafficAfterAgentIsStarted"
+
+        Write-Host "======> Given Extension is running"
+        Clear-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+        Initialize-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+        Assert-ExtensionIsRunning -Session $Session -TestConfiguration $TestConfiguration
+
+        New-AgentConfigFile -Session $Session -TestConfiguration $TestConfiguration
+
+        Write-Host "======> When Agent is started"
+        Enable-VRouterAgent -Session $Session -ConfigFilePath $TestConfiguration.AgentConfigFilePath
+        Assert-AgentIsRunning -Session $Session
+        Start-Sleep -Seconds 15  # Wait for KSync
+
+        Write-Host "======> Then Pkt0 is injected"
+        Assert-IsPkt0Injected -Session $Session
+
+        Write-Host "======> Then Pkt0 has traffic"
+        Assert-Pkt0HasTraffic -Session $Session
+
+        Write-Host "======> Then Gateway ARP was resolved through Pkt0"
+        Assert-IsGatewayArpResolvedInAgent -Session $Session
```

Comment:
I don't like that we have multiple asserts here, since there's a possible risk that one of the asserts will break another (it's a kind of test-smell). Moreover, the test name is `Pkt0ReceivesTrafficAfterAgentIsStarted` which should have nothing to do with `IsPkt0Injected` and `IsGatewayArpResolvedInAgent` - these are separate test cases.

The issue isn't visible here yet, since there aren't that many tests, but we should think about some way of grouping up tests which have a common setup/teardown procedures together and running each one separately.

The easiest way I see is having a loop and an array of functions. Something like this seems to work:

```
$a = @( (Get-Item "function:Get-NetIPAddress"), (Get-Item "function:Get-NetAdapter"))
foreach ($f in $a) { . $f }
```

so why not:

```
$testCases = @((Get-Item "function:Assert-IsPkt0Injected"),(Get-Item "function:Assert-Pkt0HasTraffic"),(Get-Item "function:Assert-IsGatewayArpResolvedInAgent"))
foreach ($testCase in $testCases) {
   # Given ...
   # When...
   # Then
   . $testCase
}
```